libtenzir/include/tenzir/split_at_string.hpp

```diff
@@ -0,0 +1,72 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2025 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "tenzir/chunk.hpp"
+#include "tenzir/generator.hpp"
+
+#include <optional>
+#include <string_view>
+
+namespace tenzir {
+
+inline auto
+split_at_string(std::string_view separator, bool include_separator = false) {
+  return [separator, include_separator](generator<chunk_ptr> input) mutable
+           -> generator<std::optional<std::string_view>> {
+    auto buffer = std::string{};
+    for (auto&& chunk : input) {
+      if (not chunk or chunk->size() == 0) {
+        co_yield std::nullopt;
+        continue;
+      }
+      buffer.append(reinterpret_cast<const char*>(chunk->data()),
+                    chunk->size());
+      std::size_t current = 0;
+      std::size_t pos = 0;
+      while ((pos = buffer.find(separator, current)) != std::string::npos) {
+        // Don't yield when we reached the end as a longer match could be found
+        // with the subsequent characters at the beginning of the next chunk.
+        if (pos + separator.size() == buffer.size()) {
+          break;
+        }
+        if (include_separator) {
+          co_yield std::string_view{buffer.data() + current,
+                                    pos + separator.size() - current};
+        } else {
+          co_yield std::string_view{buffer.data() + current, pos - current};
+        }
+        current = pos + separator.size();
+      }
+      buffer = buffer.substr(current);
+      co_yield std::nullopt;
+    }
+    if (not buffer.empty()) {
```

Comment:
There is a slight inconsistency here:
When there is no data after the last separator we don't yield an empty final string.
However, in case there is no data between the beginning and the first separator, or when we get 2 subsequent separators we yield an empty string.

This should be documented.
Alternatively you could also add an `emit_empty: bool` option and default it to false, but I guess that would be a bit much.

Reply:
I documented this behavior in dcd750f8a51f9623793619023a4867c926a456c9.